src/effects/Reverb.cpp

```diff
@@ -183,31 +190,43 @@ bool EffectReverb::ProcessInitialize(
                     mParams.mToneLow,
                     mParams.mToneHigh,
                     BLOCK,
-                    mP[i].wet);
+                    state.mP[i].wet);
    }
 
    return true;
 }
 
 bool EffectReverb::ProcessFinalize()
 {
-   for (unsigned int i = 0; i < mNumChans; i++)
+   return InstanceDeinit(mMaster);
+}
+
+bool EffectReverb::InstanceDeinit(ReverbState& state)
+{
+   for (unsigned int i = 0; i < state.mNumChans; i++)
    {
-      reverb_delete(&mP[i].reverb);
+      reverb_delete(&state.mP[i].reverb);
    }
 
-   free(mP);
+   free(state.mP);
```

Comment:
Then the above becomes unnecessary (and, wrongly double-freeing!) for avoidance of memory leaks.  The smart pointer does all.

Reply:
The code above that line had to be removed too, because the reverb_delete now happens in the destructor of `Reverb_priv_ex`